lib/Sema/TypeCheckStmt.cpp

```diff
@@ -767,7 +767,8 @@ class StmtChecker : public StmtVisitor<StmtChecker, Stmt*> {
       // statement, produce a more specific error.
       if (S->getTargetName().empty() && !ActiveLabeledStmts.empty() &&
           (isa<IfStmt>(ActiveLabeledStmts.back()) ||
-           isa<DoStmt>(ActiveLabeledStmts.back())))
+           isa<DoStmt>(ActiveLabeledStmts.back()) ||
+           isa<GuardStmt>(ActiveLabeledStmts.back())))
```

Comment:
@rintaro I've updated the code to handle the case and added your example into the test cases.

@tkremenek I prefer to submit another pull request to change the format of diagnosis message. I am also in favor of quoting `if` and `do`, but touching the message in this commit would make some noises on the logic change behind this commit.